test/basics.js

```diff
@@ -873,4 +873,28 @@ describe('Basics', () => {
       })
     })
   })
+
+  it('options should be passed', (done) => {
+    // this is a garbage test that tests whether the properties are
+    // passed on actual values are wrong and would explode
+    const opts = { test_opts: true }
+    const props = ['ackTimeout', 'connectTimeout', 'discoverPrefix', 'maxPubAcksInflight', 'nc',
+      'maxReconnectAttempts', 'stanEncoding', 'stanMaxPingOut', 'stanPingInterval', 'encoding',
+      'maxPingOut', 'maxReconnectAttempts', 'name', 'nkey', 'noRandomize', 'nonceSigner', 'pass',
+      'pedantic', 'pingInterval', 'reconnect', 'reconnectTimeWait', 'servers', 'tls', 'token',
+      'tokenHandler', 'url', 'useOldRequestStyle', 'user', 'userCreds', 'userJWT', 'verbose',
+      'waitOnFirstConnect', 'yieldTime']
+
+    props.forEach((v) => {
+      opts[v] = nuid.next()
+    })
+
+    // the property 'test_opts' aborts after parse
```

Comment:
Would it make more sense then to simply test parseOptions? Is that possible or not (as if the function is private to the lib)

Reply:
cannot - the function is a method of the object which when created calls this function.

Reply:
ok